### PR TITLE
[Feature] Expose api to trigger out-of-schedule delta snapshot

### LINKED ISSUE #114: Expose an API to trigger delta snapshot
# Story
As a developer, I want to be able to trigger a delta snapshot of etcd, on demand. Triggering via the API should be controlled via a flag  

# Motivation
The feature will be handy to run certain tests in the CI/CD pipeline, not having to wait for the scheduled time. Controlling the schedule for triggering the delta snapshot is not always an option.    

# Acceptance Criteria
- [ ] Should be able to trigger an on-demand delta snapshot

# Definition of Done 
- [ ] Knowledge is distributed: Have you spread your knowledge in pair programming/code review?
- [ ] Unit tests are provided: Have you written automated unit tests?
- [ ] Integration tests are provided: Have you written automated integration tests?
- [ ] Minimum API exposure: If you have added/changed public API, was it really necessary/is it minimal?
- [ ] Operations guide: Have you updated the operations guide about ops-relevant changes?
- [ ] User documentation: Have you updated the READMEs/docs/how-tos about user-relevant changes?

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #114

**Special notes for your reviewer**:

**Release note**:

```improvement user
Expose http API to trigger out-of-schedule delta snapshot.
```

### CODE REVIEW

**Comment by swapnilgm** [comment]
@shreyas-s-rao PTAL.

**Comment by swapnilgm** [comment]
> Can we add a test case for this please? 

Done

File: pkg/snapshot/snapshotter/snapshotter.go [hunk 1]
```diff
@@ -144,9 +147,25 @@ func (ssr *Snapshotter) TriggerFullSnapshot(ctx context.Context) error {
...
+// trigger delta snapshot externally out of regular schedule.
+func (ssr *Snapshotter) TriggerDeltaSnapshot() error {
+	ssr.SsrStateMutex.Lock()
+	defer ssr.SsrStateMutex.Unlock()
+
+	if ssr.SsrState != SnapshotterActive {
+		return fmt.Errorf("snapshotter is not active")
+	}
+	if ssr.config.deltaSnapshotInterval < deltaSnapshotIntervalThreshold {
+		return fmt.Errorf("Found delta snapshot interval %s less than 1 second. Delta snapshotting is disabled. ", ssr.config.deltaSnapshotInterval)
```
Comment (shreyas-s-rao) [inline pkg/snapshot/snapshotter/snapshotter.go]:
```suggestion
		return fmt.Errorf("Found delta snapshot interval %s less than %v. Delta snapshotting is disabled. ", ssr.config.deltaSnapshotInterval, time.Duration(deltaSnapshotIntervalThreshold))
```

File: pkg/snapshot/snapshotter/snapshotter_test.go [hunk 2]
```diff
@@ -284,6 +284,38 @@ var _ = Describe("Snapshotter", func() {
 							Expect(snapshot.Kind).ShouldNot(Equal(snapstore.SnapshotKindDelta))
 						}
 					})
+
+					It("should fail on triggering out-of-schedule full snapshot", func() {
```
Comment (shreyas-s-rao) [inline pkg/snapshot/snapshotter/snapshotter_test.go]:
```suggestion
					It("should fail on triggering out-of-schedule delta snapshot", func() {
```

File: pkg/snapshot/snapshotter/snapshotter_test.go [hunk 3]
```diff
@@ -306,7 +339,7 @@ var _ = Describe("Snapshotter", func() {
 								etcdUsername,
 								etcdPassword)
 							snapshotterConfig, err := NewSnapshotterConfig(
-								schedule,
+								fmt.Sprintf("59 %d * * *", currentHour+1), // This make sure that full snapshot timer doesn't trigger full snapshot.
```
Comment (shreyas-s-rao) [inline pkg/snapshot/snapshotter/snapshotter_test.go]:
```suggestion
								fmt.Sprintf("59 %d * * *", (currentHour+1)%24), // This make sure that full snapshot timer doesn't trigger full snapshot.
```

To handle case when `currentHour` is 23.

File: Makefile [hunk 4]
```diff
@@ -61,7 +61,7 @@ check:
 
 .PHONY: test
 test:
-	@.ci/unit_test
+	@env COVER=true .ci/unit_test
```
[the remark this answers is not in the document]
Comment (swapnilgm) [reply Makefile]: Ahh will revert it.

**Pull request merged**